daemon/container.go

```diff
@@ -326,6 +326,14 @@ func (container *Container) Start() (err error) {
 		return nil
 	}
 
+	if container.RemovalInProgress {
```

Comment:
We can probably condense these two errors into one because in terms of the end user, once a container is marked for removal at all it can't be used anyway (regardless if it got to `dead` state or not).

So:
```
if container.RemovalInProgress || container.Dead {
    return fmt.Errorf("container is marked for removal and cannot be started")
}
```